src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,371 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.accounts.AccountsType;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.model.filing.Data;
+import uk.gov.companieshouse.api.accounts.model.filing.Filing;
+import uk.gov.companieshouse.api.accounts.model.filing.Link;
+import uk.gov.companieshouse.api.accounts.model.ixbrl.Account;
+import uk.gov.companieshouse.api.accounts.service.FilingService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.accountsbuilder.AccountsBuilder;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.DocumentGeneratorConnection;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.IxbrlGenerator;
+import uk.gov.companieshouse.document.data.DocumentDescriptionHelper;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+
+@Service
+public class FilingServiceImpl implements FilingService {
+
+    private static final String API_KEY_ENV_VAR = "CHS_API_KEY";
+    private static final String DISABLE_IXBRL_VALIDATION_ENV_VAR = "DISABLE_IXBRL_VALIDATION";
+    private static final String DOCUMENT_BUCKET_NAME_ENV_VAR = "DOCUMENT_BUCKET_NAME";
+    private static final String DOCUMENT_RENDER_SERVICE_END_POINT = "/document-render/store";
+    private static final String DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR = "DOCUMENT_RENDER_SERVICE_HOST";
+    private static final String IXBRL_LOCATION = "s3://%s/%s/%s";
+    private static final String LINK_RELATIONSHIP = "accounts";
+    private static final String PERIOD_END_ON = "period_end_on";
+
+    private final EnvironmentReader environmentReader;
+    private final ObjectMapper objectMapper;
+    private final IxbrlGenerator ixbrlGenerator;
+    private final AccountsBuilder accountsBuilder;
+
+    @Autowired
+    public FilingServiceImpl(
+        EnvironmentReader environmentReader,
+        ObjectMapper objectMapper,
+        IxbrlGenerator ixbrlGenerator,
+        AccountsBuilder accountsBuilder) {
```

Comment:
Autowiring in the accountsBuilder won't work when you need to support AbridgedAccountsBuilder, SmallFullAccountsBuilder, etc.